src/contract/compiler.ts

```diff
@@ -0,0 +1,133 @@
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES WITH
+ *  REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF MERCHANTABILITY
+ *  AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY SPECIAL, DIRECT,
+ *  INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES WHATSOEVER RESULTING FROM
+ *  LOSS OF USE, DATA OR PROFITS, WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR
+ *  OTHER TORTIOUS ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR
+ *  PERFORMANCE OF THIS SOFTWARE.
+ */
+
+/**
+ * ContractCompilerHttp module
+ *
+ * @module @aeternity/aepp-sdk/es/contract/compiler
+ * @export ContractCompilerHttp
+ * @example import { ContractCompilerHttp } from '@aeternity/aepp-sdk'
+ */
+
+import { RestError } from '@azure/core-rest-pipeline'
+import stampit from '@stamp/it'
+import semverSatisfies from '../utils/semver-satisfies'
+import { Compiler as CompilerApi } from '../apis/compiler'
+import { UnsupportedVersionError } from '../utils/errors'
+
+interface ErrorFormatterBody {
+  reason?: object
+  info?: object
+  parameter?: object
+}
+interface ErrorFormatterError {
+  message: string
+  pos: {col: number, line: number}
+  type: string
+  context?: string
+}
+
+export abstract class _ContractCompilerHttp {
+  compilerApi: CompilerApi
+  compilerVersion: string
+
+  // TODO: replace with constructor after dropping account stamps
+  init ({ compilerUrl, ignoreVersion }: {
+    compilerUrl: string
+    ignoreVersion?: boolean }): void {
+    if (compilerUrl == null) return
+    this.setCompilerUrl(compilerUrl, { ignoreVersion })
+  }
+
+  setCompilerUrl (compilerUrl: string, { ignoreVersion = false } = {}): void {
+    this.compilerApi = new CompilerApi(compilerUrl, {
+      allowInsecureConnection: true,
+      additionalPolicies: [{
+        policy: {
+          name: 'version-check',
+          async sendRequest (request, next) {
+            if (ignoreVersion || new URL(request.url).pathname === '/api-version') return await next(request)
+            const args: [string, string, string] =
+               [await versionPromise, COMPILER_GE_VERSION, COMPILER_LT_VERSION]
+            if (!semverSatisfies(...args)) throw new UnsupportedVersionError('compiler', ...args)
+            return await next(request)
+          }
+        },
+        position: 'perCall'
+      }, {
+        policy: {
+          name: 'error-formatter',
+          async sendRequest (request, next) {
+            try {
+              return await next(request)
+            } catch (error) {
+              if (!(error instanceof RestError)) throw error
+              let body: ErrorFormatterError[] | ErrorFormatterBody
+              try {
+                body = JSON.parse(error.response?.bodyAsText ?? '')
+              } catch (e) {
+                throw error
+              }
+              body = body as ErrorFormatterBody
+              error.message = `${new URL(error?.request?.url ?? '').pathname.slice(1)} error`
```

Comment:
URL constructor with empty string throws en exception, I wouldn't process error at all if url is not present